aten/src/ATen/native/cuda/Distributions.cu

```diff
@@ -516,4 +540,60 @@ Tensor& capped_random_cuda_(Tensor& self, int64_t to, Generator* gen) {
   return clamped_random_cuda_(self, 0, to, gen);
 }
 
+Tensor& normal_cuda_(Tensor& self, double mean, double std, Generator* gen) {
+  AT_CHECK(std > 0.0, "normal_ expects std > 0.0, but found std=", std);
+  auto iter = TensorIterator::nullary_op(self);
+  normal_kernel_cuda(*iter, mean, std, gen);
+  return self;
+}
+
+Tensor& normal_out_cuda(Tensor& output, const Tensor& mean_, double std, Generator* gen) {
+  auto mean = std::get<0>(expand_inplace(output, mean_.to(kCUDA)));
+  normal_cuda_(output, 0, std, gen);
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(output.scalar_type(), "normal_out_cuda_cadd", [&] {
+    at::add_out(output, output, mean, static_cast<scalar_t>(1));
+   });
+  return output;
+}
+
+Tensor& normal_out_cuda(Tensor& output, double mean, const Tensor& std_, Generator* gen) {
+  auto std = std::get<0>(expand_inplace(output, std_.to(kCUDA)));
+  normal_cuda_(output, 0, 1, gen);
+  at::mul_out(output, output, std);
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(output.scalar_type(), "normal_out_cuda_cadd", [&] {
+    auto ones = at::ones_like(output);
```

Comment:
In c++, you can use add variant with Scalar `other` https://github.com/pytorch/pytorch/blob/eaa14f5f59219117973202ebb79e96b960a92f9b/aten/src/ATen/native/native_functions.yaml#L138. In any case, creating full-size ones is wasteful, you could broadcast a single element ones tensor.

Reply:
Also, I think addcmul would allow you to have a single pointwise operation here instead of separate mul and add.